internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -200,15 +200,21 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 		log.WithError(err).Error("error setting owner reference")
 		return ctrl.Result{Requeue: true}, err
 	}
+	pullSecret, err := r.getPullSecret(ctx, clusterDeployment, clusterInstall)
+	if err != nil {
+		log.WithError(err).Errorf("failed to get pull secret for cluster deployment %s", clusterDeployment.Name)
+		return r.updateStatus(ctx, log, clusterInstall, clusterDeployment, nil, err)
+	}
+	releaseImage, err := r.getReleaseImage(ctx, clusterDeployment, clusterInstall, pullSecret)
```

Comment:
Good question. I wasn't sure if release images are required for day2/installed clusters. Do you know if it's needed?

In the function, I do a best attempt to get the image: if it's specified, regardless of the cluster status, we'll try to get it. However, I won't throw an error if it's day2 because I thought it wasn't required for this flow. 

Do you think it's best to always just require the release image to be specified and to always try and retrieve it?